test/functionality/dataflow/processing-of-elements/atomic/dataflow-redefine-tests.ts

```diff
@@ -0,0 +1,146 @@
+		shell, `\`if\` <- function(...) 2
+if(1) 
+   x <- 3
+print(x)`, emptyGraph()
+			.use('14', 'x')
+			.reads('14', '8')
+			.call('6', '<-', [argumentInCall('0'), argumentInCall('5')], { returns: ['0'], reads: [BuiltIn] })
+			.argument('6', ['5', '0'])
+			.call('10', '<-', [argumentInCall('8'), argumentInCall('9')], { returns: ['8'], reads: [BuiltIn], environment: defaultEnv().defineFunction('if', '0', '6') })
+			.argument('10', ['9', '8'])
+			.definesOnCall('10', '1')
+			.argument('12', '10')
+			.call('12', 'if', [argumentInCall('7'), argumentInCall('10'), EmptyArgument], { returns: ['3'], reads: ['0'], environment: defaultEnv().defineFunction('if', '0', '6') })
+			.argument('12', '7')
+			.calls('12', '5')
+			.argument('16', '14')
+			.call('16', 'print', [argumentInCall('14')], { returns: ['14'], reads: [BuiltIn], environment: defaultEnv().defineFunction('if', '0', '6').defineVariable('x', '8', '10') })
+			.defineVariable('1', '...', { definedBy: [] }, false)
+			.constant('3', undefined, false)
+			.defineFunction('5', ['3'], {
+				out:               [],
+				in:                [{ nodeId: '3', name: undefined, controlDependencies: [] }],
+				unknownReferences: [],
+				entryPoint:        '3',
+				graph:             new Set(['1', '3']),
+				environment:       defaultEnv().pushEnv().defineParameter('...', '1', '2')
+			})
+			.defineVariable('0', '`if`', { definedBy: ['5', '6'] })
+			.constant('7')
+			.definesOnCall('7', '1')
+			.constant('9')
+			.defineVariable('8', 'x', { definedBy: ['9', '10'] }))
+	assertDataflow(label('<-', []),
+		shell, `x <- 2
+\`<-\` <- \`*\`
+x <- 3
+print(y = x)`, emptyGraph()
+			.use('4', '`*`')
+			.reads('4', BuiltIn)
+			.use('6', 'x')
+			.reads('6', '0')
+			.use('11', 'x')
+			.reads('11', '0')
+			.use('12', 'y')
+			.reads('12', '11')
+			.call('2', '<-', [argumentInCall('0'), argumentInCall('1')], { returns: ['0'], reads: [BuiltIn] })
+			.argument('2', ['1', '0'])
+			.argument('5', '4')
+			.call('5', '<-', [argumentInCall('3'), argumentInCall('4')], { returns: ['3'], reads: [BuiltIn], environment: defaultEnv().defineVariable('x', '0', '2') })
+			.argument('5', '3')
+			.argument('8', '6')
+			.call('8', '<-', [argumentInCall('6'), argumentInCall('7')], { returns: [], reads: ['3'], environment: defaultEnv().defineVariable('x', '0', '2').defineVariable('<-', '3', '5') })
+			.argument('8', '7')
+			.argument('13', '12')
+			.call('13', 'print', [argumentInCall('12', { name: 'y' } )], { returns: ['12'], reads: [BuiltIn], environment: defaultEnv().defineVariable('x', '0', '2').defineVariable('<-', '3', '5') })
+			.constant('1')
+			.defineVariable('0', 'x', { definedBy: ['1', '2'] })
+			.defineVariable('3', '`<-`', { definedBy: ['4', '5'] })
+			.constant('7'))
+	assertDataflow(label('<- in function', ['name-normal', ...OperatorDatabase['<-'].capabilities, 'normal-definition', 'implicit-return','newlines', ...OperatorDatabase['*'].capabilities, 'call-normal', 'unnamed-arguments']),
```

Comment:
super unsure about this one, feels like it should contain `*` as an `RBinaryOp`, but only contains the `RSymbol` of `\`*\`` @EagleoutIce

Reply:
It should contain it as a binary op. By redefining `<-` as `*` we execute it in line 101.